data/mods/gen2/scripts.ts

```diff
@@ -560,35 +559,59 @@ export const Scripts: ModdedBattleScriptsData = {
 			basePower = this.battle.clampIntRange(basePower, 1);
 
 			// We now check for attacker and defender
-			let level = pokemon.level;
+			let level = source.level;
 
 			// Using Beat Up
 			if (move.allies) {
-				this.battle.add('-activate', pokemon, 'move: Beat Up', '[of] ' + move.allies[0].name);
+				this.battle.add('-activate', source, 'move: Beat Up', '[of] ' + move.allies[0].name);
 				level = move.allies[0].level;
 			}
 
-			let attacker = pokemon;
-			const defender = target;
-			if (move.useTargetOffensive) attacker = target;
-			let atkType: StatIDExceptHP = (move.category === 'Physical') ? 'atk' : 'spa';
-			const defType: StatIDExceptHP = (move.defensiveCategory === 'Physical') ? 'def' : 'spd';
-			if (move.useSourceDefensiveAsOffensive) atkType = defType;
+			const attacker =
+				(move.useBaseOffensiveStatAndBoosts && move.useBaseOffensiveStatAndBoosts.includes('target')) ? target : source;
+			const defender =
+				(move.useBaseDefensiveStatAndBoosts && move.useBaseDefensiveStatAndBoosts.includes('source')) ? source : target;
+
+			let atkType: AllStatIDs = move.category === 'Physical' ? 'atk' : 'spa';
+			if (move.useBaseOffensiveStatAndBoosts) {
+				atkType = move.useBaseOffensiveStatAndBoosts.substr(7) as AllStatIDs;
+			}
+			let defType: AllStatIDs = move.category === 'Physical' ? 'def' : 'spd';
+			if (move.useBaseDefensiveStatAndBoosts) {
+				defType = move.useBaseDefensiveStatAndBoosts.substr(7) as AllStatIDs;
+			}
 			let unboosted = false;
 			let noburndrop = false;
 
 			if (isCrit) {
 				if (!suppressMessages) this.battle.add('-crit', target);
-				// Stat level modifications are ignored if they are neutral to or favour the defender.
+				// Stat level modifications are ignored if they are neutral to or favour) the defender.
```

Comment:
?

Reply:
No idea why that is there or how I missed that

> I think the changes from `pokemon` to `source` should have been done separately from this, if at all. It makes it very annoying to find what's actually changing here. I like that the huge amount of code duplication you had is gone, at least.
> 
> Not a big fan of "future-proofing" past gens, but I understand why you did it like this. What's the argument for not refactoring all of that stuff out later? OMs can maybe use it?
> 
> Tentatively approving, subject to DaWoblefet's upcoming Wonder Room test passing and whichever Smogon Tournament is in progress finishing. Would like @Zarel to give this a look over meanwhile, but I'll merge after those conditions otherwise. 👍

Consistency. I wanted them all to use the same code when possible. Sorry about including the pokemon -> source change, I thought it made the code more readable. I will try and segment my changes more in the future.

Reply:
Oops. I will fix that in the next commit. Nice catch

Reply:
fixed